src/ChunkMap.h

```diff
@@ -460,15 +468,25 @@ class cChunkMap
 		/** Returns true if there is an entity with the specified ID within this layer's chunks */
 		bool HasEntity(UInt32 a_EntityID);
 
+		/** Returns how many chunks this layer has. */
+		size_t GetNumChunks();
+
 	protected:
 
 		cChunkPtr m_Chunks[LAYER_SIZE * LAYER_SIZE];
 		int m_LayerX;
 		int m_LayerZ;
 		cChunkMap * m_Parent;
-		int m_NumChunksLoaded;
 
 		cAllocationPool<cChunkData::sChunkSection> & m_Pool;
+	private:
+		/** Increase the internal chunk counter. */
+		void IncreaseChunkCounter();
+
+		/** Decrease the internal chunk counter. */
+		void DecreaseChunkCounter();
+
+		size_t m_ChunkCounter;
```

Comment:
>I have an issue with the name. Counter is something (an object, most likely) that does some complex counting. NumChunks would be a preferred name.

Ok.

>This is what you said with regards to ParentChunk, IsTicking and IsChangingWorlds and is what I'm saying here.

ParentChunk, IsTicking and IsChangingWorlds are redundant and do the same thing. There is no redundancy here, there's just caching of an expensive operation. It's not the same.

>each with their updater functions which need to be remembered to be called

The one that bother me is the m_IsDirty counting at cChunk. That one does seem very error prone and I have to modify it.

As for the rest, I agree they're a bit ugly, but they should be very easy to keep track of: There isn't much room for error in: `increment 1 on creation, decrement 1 on destruction` or `increment X  RAM on creation, remove X RAM on creation`.